src/dsp_tools/utils/connection_live.py

```diff
@@ -352,6 +341,8 @@ def _try_network_action(self, action: Callable[[], Response]) -> Response:
                 self._log_and_sleep(reason="Timeout Error", retry_counter=i)
                 continue
             except (ConnectionError, RequestException):
+                self.session.close()
+                self.session = requests.Session()
```

Comment:
Superficial comment here. But if we want to keep with the new style of having the functions after they are used. _log_and_sleep should be moved down.

Reply:
I will do that in the next PR which is already in the pipeline. (I don't want to create too many merge conflicts)